toml/decoder.py

```diff
@@ -957,3 +964,54 @@ def load_array(self, a):
                     atype = ntype
                 retval.append(nval)
         return retval
+
+    def handle_inline_comment(self, line, currentlevel):
+        return line
+
+    def handle_comment(self, currentlevel, comment_no, recent, line):
+        return comment_no
+
+    def comment_lint(self, sl, i, beginline):
```

Comment:
Why is beginline a necessary parameter and why does this method return it?

Reply:
This part of the code removes the comment data if TomlPreseveCommentDecoder is not used. beginline may not make sense if TomlPreseveCommentDecoder is not used. 

But if TomlPreseveCommentDecoder is used, the comment data has to be preserved and beginline must be set to false for the decoder to not misinterpret the comment.